### DISCUSSION TITLE
Compatibility Issues with cuQuantum benchmark 0.3 and Latest Versions of Qiskit and PennyLane

### DISCUSSION THREAD

**Discussion by Quleaf** [NVIDIA/cuQuantum#146] [Q&A]
Hello Benchmark Team,

I have been using the cuQuantum benchmark version 0.3, released a year ago. Recently, I noticed that it is no longer compatible with the latest versions of Qiskit and PennyLane. The primary issue seems to stem from changes in the API calls of these frameworks.

Would you happen to have plans to release a new version of the benchmark in the near future to address these compatibility issues? We heavily rely on these benchmarks to evaluate and optimize the performance of our quantum algorithms and would like to continue using cuQuantum for our research.

**Comment by yangcal** [comment]
Yes we're actively maintaining cuquantum-benchmarks. Can you share the command line that causes issue? It would be easier for us to look into it if you could also share the versions of qiskit/pennylane that you're using.

**Reply by Quleaf** [reply]
I'd like to provide a specific example that illustrates the compatibility issue. I'm using Qiskit version 1.0.2 and Qiskit Aer version 0.14.2. When I execute the command:
`python -m cuquantum_benchmarks circuit --benchmark ghz --frontend qiskit --backend aer --nqubits 20`

 encounter the following error:
`line 48, in find_version
    if hasattr(qiskit_aer, "__version__"):
NameError: name 'qiskit_aer' is not defined`

Upon inspecting the source code at [this location](https://github.com/NVIDIA/cuQuantum/blob/main/benchmarks/cuquantum_benchmarks/backends/backend_qiskit.py) in your GitHub repository, I noticed that on line 14, only `qiskit` is imported. It appears necessary to also `import qiskit_aer` to resolve this issue. Adding import qiskit_aer in the relevant sections of the code could potentially fix this compatibility problem. 

Could you please look into this? Thank you for your attention to this matter.

**Reply by Quleaf** [reply]
I forgot to mention, the fundamental reason for this issue is that after version 1.0.0, qiskit_aer was separated from qiskit, and the backend did not account for this change. The link to the related change notice is as follows: https://docs.quantum.ibm.com/api/migration-guides/qiskit-1.0-features.

**Reply by yangcal** [reply]
Thanks for reporting. We're actually aware of this issue and have been working on a fix. Please stay tuned for our next release.

**Comment by Quleaf** [comment]
Additionally, the interface `pennylane.device` in PennyLane underwent changes quite some time ago, where it no longer (Sep 2023) accepts the `c_dtype` parameter. The specific commit detailing this change can be found here: https://github.com/PennyLaneAI/pennylane/commit/c6ef610bcdd59188a2b8c3e56cc9b6cd5aeb858c. As a result, when I run the command:
`python -m cuquantum_benchmarks circuit --benchmark ghz --frontend pennylane --backend pennylane --nqubits 20`
I encountered the following error: "TypeError: DefaultQubit.init() got an unexpected keyword argument 'c_dtype'". According to the benchmark's commit history, there have been no updates since the release of version 0.3.0.

**Reply by Quleaf** [reply]
If possible, I highly recommend locking the version of dependencies (such as qiskit-aer, pennylane, etc.) of the cuquantum benchmark in setup.py.
